### PR TITLE
Add option to retrieve oauthtoken from a password store

### PR DESCRIPTION
**Pull request by eonpatapon**
In order to avoid leaving plaintext password in config files

### CODE REVIEW

File: git-hub [hunk 1]
```diff
@@ -341,6 +341,14 @@ class Config:
     def __init__(self):
         self.username = git_config('username', getpass.getuser())
         self.oauthtoken = git_config('oauthtoken')
+        self.oauthtokencmd = git_config('oauthtokencmd')
+        if self.oauthtokencmd is not None:
+            try:
+                self.oauthtoken = subprocess.check_output(
+                        self.oauthtokencmd.split()).strip()
```
Comment (jwilk) [inline git-hub]: Use `shlex.split()` for command-line splitting.
Or, better, pass the whole command line to shell:
```python
subprocess.check_output(self.oauthtokencmd, shell=True)
```

File: man.rst [hunk 2]
```diff
@@ -460,6 +460,10 @@ from. These are the git config keys used:
   required, you shouldn't need to set this variable manually. Use the `setup`
   command instead.
 
+`hub.oauthtokencmd`
+  A command to retrieve the oauth token from a password store.
+  When set `hub.oauthtoken` isn't needed.
```
Comment (jwilk) [inline man.rst]:
```suggestion
  When set, `hub.oauthtoken` isn't needed.
```

File: git-hub (hunk 1 above)
Comment (eonpatapon) [reply git-hub]: Done

File: man.rst (hunk 2 above)
Comment (eonpatapon) [reply man.rst]: Done

**Comment by llucax** [comment]
Hi @eonpatapon, I don't know how I missed this PR, but I really like it. Can you add some release notes too? Just add a new file named something like `relnotes/oauthcmd.feature.md` (you can see the existing files as examples or read the [full docs on how to write it](https://github.com/sociomantic-tsunami/neptune/blob/v0.x.x/doc/library-contributor.rst#release-notes)).

Also, I think I would rather reuse the `oauthtoken` configuration variable and make it run a command when it starts with `!`, which is a very common idiom in many places in the unix world. Would it be too much to change that too? :grimacing:

**Comment by eonpatapon** [comment]
hi, actually it's been a while and I don't use this tool anymore because it didn't really fit my workflow

but if the idea seems good to you feel free to use the code in this PR if you want

**Comment by llucax** [comment]
@eonpatapon fair enough, I might pick it up and include it then.  Thanks!

**Comment by llucax** [comment]
Superseded by #283. Thanks!

**Pull request closed without merging**